### ISSUE TITLE
downloading the wrong version of a package didn't error out

### ISSUE THREAD

**Issue by patrickhuber** [patrickhuber/wrangle#12]
On linux, with the following configuration. Install package only failed on the extract step. Need to investigate why the download worked.

```
- name: wrangle
  version: 0.2.1
  platforms:
  - name: linux
    alias: wrangle
    download:
      url: https://github.com/patrickhuber/wrangle/releases/download/((version))/wrangle-linux-amd64.tgz
      out: wrangle-((version))-linux-amd64.tgz
    extract:
      filter: wrangle-linux-amd64
      out: wrangle-((version))-linux-amd64
  - name: windows
    alias: wrangle.exe
    download:
      url: https://github.com/patrickhuber/wrangle/releases/download/((version))/wrangle-windows-amd64.tgz
      out: wrangle-((version))-windows-amd64.tgz
    extract:
      filter: wrangle-windows-amd64.exe
      out: wrangle-((version))-windows-amd64.exe
  - name: darwin
    alias: wrangle
    download:
      url: https://github.com/patrickhuber/wrangle/releases/download/((version))/wrangle-darwin-amd64.tgz
      out: wrangle-((version))-darwin-amd64.tgz
    extract:
      filter: wrangle-darwin-amd64
      out: wrangle-((version))-darwin-amd64
```

**Comment by patrickhuber** [comment]
It appears to be an issue with 404. 

Redirects should probably be followed (302, 301, 307 etc)

Any non 300 or 200 should throw an error instead of attempting a download

**Comment by patrickhuber** [comment]
fixed in v0.4.2
